src/crypto-api/CryptoEvent.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export enum CryptoEvent {
```

Comment:
I think having a completely separate set of events is somewhat confusing.

Your definition of the new `CryptoEvent` here is good, but instead of duplicating the entries, we could redefine the old `CryptoEvent` in terms of the new one:


```typescript
// in src/crypto/index.ts
import {CryptoEvent as CryptoApiCryptoEvent} from "../crypto-api/index.ts";

export enum CryptoEvent {
    UserTrustStatusChanged = CryptoApiCryptoEvent.UserTrustStatusChanged,
    // ...
}
```

Similarly, you can make the old `CryptoEventHandlerMap` extend the new one.

Reply:
According to https://sonarcloud.io/organizations/matrix-org/rules?open=typescript%3AS6550&rule_key=typescript%3AS6550 this is breaking this rule. We can only use literals with our rules.

Reply:
Can we just dismiss the sonarcloud complaint? After all, the only reason we're continuing to keep `UserTrustStatusChanged` in the (old) `CryptoEvent` enum is as a backwards-compatibility re-export from the new enum.